ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezTask.java

```diff
@@ -674,6 +687,13 @@ DAGClient submit(DAG dag, Ref<TezSessionState> sessionStateRef) throws Exception
     return new SyncDagClient(dagClient);
   }
 
+  private DAGClient submitInternal(DAG dag, TezSessionState sessionState) throws TezException, IOException {
+    TezClient tezClient = sessionState.getSession();
+    DAGClient dagClient = tezClient.submitDAG(dag);
+    runtimeContext.initFromTezClient(tezClient);
```

Comment:
wouldn't it be better to init
````
runtimeContext.initFromTezClient(sessionState.getSession())
````
before we return the SyncDagClient?

Reply:
ack, that way we don't need the extra method